### PR TITLE
Add BC Break Policy

### PR DESCRIPTION
**Pull request by samsonasik**
**Checklist:**
- [x] Securely signed commits

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -63,6 +63,11 @@ each other and from whatever else you are doing with your repository!
 
 You must [GPG-sign](./contributing/signing.rst) your work, certifying that you either wrote the work or otherwise have the right to pass it on to an open-source project. This is *not* just a "signed-off-by" commit, but instead, a digitally signed one.
 
+### BC Break Policy
+
+1. New class/property/constant in `system`, that is OK.
+2. New class/property/constant in `app` directory that will be consumed in `system` is a big no without handling in `system` itself if the new class/property/constant not exists.
+
```
Comment (MGatner) [inline CONTRIBUTING.md]:
```suggestion
### Breaking Changes

In general any changes that would disrupt existing uses of the framework are considered "breaking changes" and are not allowed. A few specific examples to pay attention to:

1. New classes/properties/constants in `system` are acceptable, but anything in the `app` directory that will be used in `system` should be backwards-compatible
2. Any changes to non-private methods must be backwards-compatible with the original definition

```

**Review by MGatner — CHANGES_REQUESTED** [review]
I suggested some more verbose and different wording

File: CONTRIBUTING.md (hunk 1 above)
Comment (samsonasik) [reply CONTRIBUTING.md]: implemented.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -63,6 +63,13 @@ each other and from whatever else you are doing with your repository!
 
 You must [GPG-sign](./contributing/signing.rst) your work, certifying that you either wrote the work or otherwise have the right to pass it on to an open-source project. This is *not* just a "signed-off-by" commit, but instead, a digitally signed one.
 
+### Breaking Changes
+
+In general any changes that would disrupt existing uses of the framework are considered "breaking changes" and are not allowed. A few specific examples to pay attention to:
```
Comment (paulbalandan) [inline CONTRIBUTING.md:68]:
```suggestion
In general, any change that would disrupt existing uses of the framework is considered a "breaking change" and will not be favorably considered. A few specific examples to pay attention to:
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -63,6 +63,13 @@ each other and from whatever else you are doing with your repository!
...
+
+1. New classes/properties/constants in `system` are acceptable, but anything in the `app` directory that will be used in `system` should be backwards-compatible.
+2. Any changes to non-private methods must be backwards-compatible with the original definition.
+
```
Comment (paulbalandan) [inline CONTRIBUTING.md:72]: More additions:
3. Deleting non-private properties or methods without prior deprecation notices is frowned upon and will likely be closed.
4. Deleting or renaming public classes and interfaces, as well as those not marked as `@internal`, without prior deprecation notices or not providing fallback solutions will also not be favorably considered.

**Review by paulbalandan — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (samsonasik) [reply CONTRIBUTING.md]: applied.

File: CONTRIBUTING.md (hunk 3 above)
Comment (samsonasik) [reply CONTRIBUTING.md:74]: applied.

**Review by michalsn — APPROVED** [review]
This is badly needed. Though I doubt that people will read this before sending a PR, at least we'll have a place to send them to for an explanation.

**Review by lonnieezell — APPROVED** [review]

**Pull request merged**